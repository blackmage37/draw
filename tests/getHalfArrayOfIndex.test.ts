import getHalfArrayOfIndex from '../src/utils/getHalfArrayOfIndex';

describe('getHalfArrayOfIndex', () => {
  describe.each([
    { arr: [0, 1, 2, 3, 4, 5, 6, 7, 8] },
    { arr: [0, 1, 2] },
    { arr: [0, 1] },
    { arr: [0] },
    { arr: [] },
  ])('%s', ({ arr }) => {
    it.each([-1, ...arr.map((_, i) => i), arr.length])('%s', i => {
      const result = getHalfArrayOfIndex(arr, i);
      expect(result).toMatchSnapshot();
    });
  });
});
