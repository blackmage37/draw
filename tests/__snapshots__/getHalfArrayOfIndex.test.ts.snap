// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getHalfArrayOfIndex > { arr: [ +0 ] } > -1 1`] = `[]`;

exports[`getHalfArrayOfIndex > { arr: [ +0 ] } > 0 1`] = `[]`;

exports[`getHalfArrayOfIndex > { arr: [ +0 ] } > 1 1`] = `[]`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1 ] } > -1 1`] = `
[
  0,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1 ] } > 0 1`] = `
[
  0,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1 ] } > 1 1`] = `
[
  1,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1 ] } > 2 1`] = `
[
  1,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2 ] } > -1 1`] = `
[
  0,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2 ] } > 0 1`] = `
[
  0,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2 ] } > 1 1`] = `
[
  1,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2 ] } > 2 1`] = `
[
  1,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2 ] } > 3 1`] = `
[
  1,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > -1 1`] = `
[
  0,
  1,
  2,
  3,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 0 1`] = `
[
  0,
  1,
  2,
  3,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 1 1`] = `
[
  0,
  1,
  2,
  3,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 2 1`] = `
[
  0,
  1,
  2,
  3,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 3 1`] = `
[
  0,
  1,
  2,
  3,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 4 1`] = `
[
  4,
  5,
  6,
  7,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 5 1`] = `
[
  4,
  5,
  6,
  7,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 6 1`] = `
[
  4,
  5,
  6,
  7,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 7 1`] = `
[
  4,
  5,
  6,
  7,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 8 1`] = `
[
  4,
  5,
  6,
  7,
]
`;

exports[`getHalfArrayOfIndex > { arr: [ +0, 1, 2, 3, 4, 5, 6, 7, 8 ] } > 9 1`] = `
[
  4,
  5,
  6,
  7,
]
`;

exports[`getHalfArrayOfIndex > { arr: [] } > -1 1`] = `[]`;

exports[`getHalfArrayOfIndex > { arr: [] } > 0 1`] = `[]`;
